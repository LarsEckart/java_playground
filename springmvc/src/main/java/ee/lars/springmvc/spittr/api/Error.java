package ee.lars.springmvc.spittr.api;

public class Error {

    private final int code;
    private final String message;

    public Error(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return this.code;
    }

    public String getMessage() {
        return this.message;
    }
}