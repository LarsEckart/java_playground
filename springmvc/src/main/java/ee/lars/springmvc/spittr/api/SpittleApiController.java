package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/spittles")
public class SpittleApiController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleApiController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }

    @RequestMapping(method = RequestMethod.POST, consumes = "application/json")
    public Spittle saveSpittle(@RequestBody Spittle spittle) {
        return spittleRepository.save(spittle);
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public ResponseEntity<?> spittleById(@PathVariable long id) {
        Spittle spittle = this.spittleRepository.findOne(id);
        if (spittle == null) {
            Error error = new Error(4, "Spittle [" + id + "] not found");
            return new ResponseEntity<Error>(error, HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<Spittle>(spittle, HttpStatus.OK);
    }
}
