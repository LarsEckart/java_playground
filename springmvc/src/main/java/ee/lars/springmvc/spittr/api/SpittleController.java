package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
@RequestMapping("/spittles")
public class SpittleController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }
}
