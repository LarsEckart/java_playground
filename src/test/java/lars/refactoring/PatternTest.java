package lars.refactoring;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator.ReplaceUnderscores;
import org.junit.jupiter.api.Test;

@DisplayNameGeneration(ReplaceUnderscores.class)
class PatternTest {

  @Test
  void doesNotChangeUntaggedText() {
    assertEquals("plain text", Pattern.formatText("plain text"));
  }

  @Test
  void removesTextBetweenAngleBracketPairs() {
    assertEquals("", Pattern.formatText("<>"));
  }

  @Test
  void approvals() {
    String[] input = new String[] {null, "", "plain text", "<>", "this is <b>important</b>"};
    Approvals.verifyAll("Stripping html tags", input, a -> testFormatText(a));
  }

  private String testFormatText(String input) {
    String result;
    try {
      result = Pattern.formatText(input);
    } catch (Exception e) {
      result = e.getClass().getName();
    }
    return input + " => " + result;
  }
}
