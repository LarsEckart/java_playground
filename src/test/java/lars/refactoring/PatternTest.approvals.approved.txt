Stripping html tags


null => java.lang.NullPointerException
 => 
plain text => plain text
<> => 
this is <b>important</b> => this is important
